File: backend/src/main/java/br/ufpr/sept/so2/modules/iam/application/ports/RefreshTokenRepository.java
package br.ufpr.sept.so2.modules.iam.application.ports;

import br.ufpr.sept.so2.modules.iam.domain.RefreshSessao;

import java.util.Optional;
import java.util.UUID;

public interface RefreshTokenRepository {

    RefreshSessao save(RefreshSessao sessao);

    Optional<RefreshSessao> lockByTokenHash(String tokenHash);

    void revokeAllByUsuarioId(UUID usuarioId);
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/iam/application/ports/RefreshTokenRepository.kt
package br.ufpr.sept.so2.modules.iam.application.ports

import br.ufpr.sept.so2.modules.iam.domain.RefreshSessao
import java.util.Optional
import java.util.UUID

interface RefreshTokenRepository {
    fun save(sessao: RefreshSessao): RefreshSessao

    fun lockByTokenHash(tokenHash: String): Optional<RefreshSessao>

    fun revokeAllByUsuarioId(usuarioId: UUID)
}
